src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -118,6 +118,12 @@ private async Task HandleClusterDatabaseChanged(string databaseName, long index,
 
                         if (rawRecord.IsSharded())
                         {
+                            if (type == nameof(DestinationMigrationConfirmCommand))
+                            {
+                                if (rawRecord.BucketMigrations.Any(m => m.Value.ConfirmationIndex == index))
+                                    _shardedDatabases.Clear();
```

Comment:
What is that about? `_shardedDatabases` is a global value, no? Why are you clearing all of it?
Also, what about disposing of `ShardedContext` ?

Reply:
What would happen on snapshot install for this?

Reply:
Assume that the mapping update happens on snapshot install.

Also, you are now removing this for _all_ databases.

I _really_ don't like this apparoach

Reply:
If I understand correctly we are talking here about 2 things
1. What happened on snapshot install -> I would say clear it all.
2. Upon migration completion -> Yes we need to update only relevant dbs (instead of clearing)
Do I get that right?